src/ucp/core/ucp_mm.c

```diff
@@ -141,12 +141,15 @@ static ucs_status_t ucp_mem_alloc(ucp_context_h context, size_t length,
          * name specified in the configuration.
          */
         num_mds = 0;
-        if (method == UCT_ALLOC_METHOD_MD) {
+        if (method == UCT_ALLOC_METHOD_MD && !(uct_flags & UCT_MD_MEM_FLAG_FIXED)) {
```

Comment:
brackets should be used for both `if ((method == UCT_ALLOC_METHOD_MD) && !(uct_flags & UCT_MD_MEM_FLAG_FIXED))`